src/UsgsAstroLsSensorModel.cpp

```diff
@@ -617,6 +617,43 @@ void UsgsAstroLsSensorModel::updateState() {
 
   // Set focal length variance
   m_covariance[15 * num_params + 15] = positionVariance;
+
+  // Test if a pixel projected to the ground and projected back
+  // returns to the original location. If not, adjust m_iTransL. It is
+  // very important here to pick a pixel which is not an integer, as
+  // this test may succeed for integer pixels and fail for non-integer
+  // ones. Also, need to pick the answer with the smallest achieved
+  // precision, even when neither of them is smaller than the desired
+  // precision.
+  
+  lineCtr = round(m_nLines / 2.0) + 0.5;
+  sampCtr = round(m_nSamples / 2.0) + 0.5;
+
+  double desiredPrecision = 0.001;
+  ip = csm::ImageCoord(lineCtr, sampCtr);
+  csm::EcefCoord xyz = imageToGround(ip, refHeight);
+
+  double sign1 = 1.0;
+  for (int it = 0; it < 3; it++) 
+    m_iTransL[it] = std::abs(m_iTransL[it]) * sign1;
```

Comment:
I'm worried about this abs. If there is a iTransL with one positive and one negative term then this will effectively "flip" the detector. I don't think that the abs is even needed.